src/Workspaces/Core/Portable/FindSymbols/ReferenceLocation.cs

```diff
@@ -54,9 +54,16 @@ namespace Microsoft.CodeAnalysis.FindSymbols
         /// </summary>
         internal ImmutableDictionary<string, string> AdditionalProperties { get; }
 
+        /// <summary>
+        /// If this reference location is within a string literal, then this property
+        /// indicates the location of the containing string literal token.
+        /// Otherwise, <see cref="Location.None"/>.
+        /// </summary>
+        internal Location ContainingStringLocation { get; }
```

Comment:
actually... why do you need this?

Reply:
This location maps to the containing string literal token with the reference and serves two purposes:
1. Differentiates these reference locations from rest of the reference locations. These locations span within a string literal token, while all other references span an entire token.
2. It nicely fits into the existing "Rename inside strings" functionality for inline rename. That feature expects a location for the entire string literal token within which rename of strings needs to be performed.